packages/loader/container-loader/src/loader.ts

```diff
@@ -357,8 +357,19 @@ export class Loader implements IHostLoader {
 		return this;
 	}
 
-	public async createDetachedContainer(codeDetails: IFluidCodeDetails): Promise<IContainer> {
-		const container = await Container.createDetached(this.services, codeDetails);
+	public async createDetachedContainer(
+		codeDetails: IFluidCodeDetails,
+		containerCreateHeaders?: IRequestHeader,
+	): Promise<IContainer> {
+		// "this.services" doesn't contain property for clientDetailsOverride
+		const container = await Container.createDetached(
+			{
+				canReconnect: containerCreateHeaders?.[LoaderHeader.reconnect],
+				clientDetailsOverride: containerCreateHeaders?.[LoaderHeader.clientDetails],
+				...this.services,
```

Comment:
this then gets simplified to 
```suggestion
				...this.props,
				...this.services,
```